recipes/powershell2.rb

```diff
@@ -61,11 +63,19 @@
 
     include_recipe 'ms_dotnet2'
 
+    powershell_version_cmd = 'powershell.exe $PSVersionTable.PSVersion.Major'
+    shell_out = Mixlib::ShellOut.new(powershell_version_cmd)
+    shell_out.run_command
+    powershell_version = shell_out.stdout.to_f
+
+    # Reboot if user specifies immediate_reboot
```

Comment:
@adamedx , we don't reboot in all cases. We reboot only if user specifies `default['powershell']['installation_reboot_mode'] == 'immediate_reboot'` in default attribute file.